File: solutions/python/structure/fixed/StaticStackADT.py
class StaticStackADT:
    def __init__(self, capacidad: int):
        self.capacidad = capacidad
        self.stack = [None] * capacidad  # lista de tamaño fijo
        self.tope = -1  # índice del último elemento

    def getElement(self) -> int:
        if self.is_empty():
            raise Exception("La pila está vacía.")
        return self.stack[self.tope]

    def add(self, value: int):
        if self.tope == self.capacidad - 1:
            raise Exception("La pila está llena.")
        self.tope += 1
        self.stack[self.tope] = value

    def remove(self):
        if self.is_empty():
            raise Exception("La pila está vacía.")
        self.stack[self.tope] = None  # opcional: limpiar la posición
        self.tope -= 1

    def is_empty(self) -> bool:
        return self.tope == -1

    def mostrar(self):
        for i in range(self.tope, -1, -1):
            print(self.stack[i], end=" -> ")
        print("None")

File: solutions/javascript/src/structure/fixed/StaticStackADT.js
import DynamicStackADT from "../dynamic/DynamicStackADT.js";

class StaticStackADT {
  constructor(capacity) {
    this.capacity = capacity;
    this.stack = new Array(capacity);
    this.top = -1;
  }

  add(value) {
    if (this.top >= this.capacity - 1) {
      throw new Error("La pila ha alcanzado su capacidad máxima");
    }
    this.stack[++this.top] = value;
  }

  remove() {
    if (this.isEmpty()) {
      throw new Error("La pila está vacía");
    }
    this.top--;
  }

  getElement() {
    if (this.isEmpty()) {
      throw new Error("La pila está vacía");
    }
    return this.stack[this.top];
  }

  isEmpty() {
    return this.top === -1;
  }
}

export default StaticStackADT;